packages/console/src/components/ProTag/index.tsx

```diff
@@ -1,19 +1,33 @@
 import classNames from 'classnames';
+import { useContext, useMemo } from 'react';
 
+import { isDevFeaturesEnabled } from '@/consts/env';
+import { TenantsContext } from '@/contexts/TenantsProvider';
 import DynamicT from '@/ds-components/DynamicT';
 
 import * as styles from './index.module.scss';
 
 type Props = {
+  isVisible: boolean;
   className?: string;
 };
 
-function ProTag({ className }: Props) {
-  return (
+function ProTag({ isVisible, className }: Props) {
+  const { isDevTenant } = useContext(TenantsContext);
+
+  const shouldDisplay = useMemo(() => {
+    if (isDevFeaturesEnabled && isDevTenant) {
+      return true;
+    }
+
+    return isVisible;
+  }, [isDevTenant, isVisible]);
+
+  return shouldDisplay ? (
     <div className={classNames(styles.tag, className)}>
       <DynamicT forKey="upsell.pro_tag" />
     </div>
-  );
+  ) : null;
```

Comment:
This is confusing. The `isVisible` prop indicates the tag should be displayed, while in fact there are extra condition checks inside the component. As a result, the show/hide status can be opposite with the `isVisible` prop...

Reply:
Updated by renaming the props name to `isVisibleInProdEnv`.